File: Creational/Object Pool/JavaScript/js/ReusableObject.js
export class ReusableObject {
    static counter = 0;

    constructor() {
        this.id = ++ReusableObject.counter;
        this.log(`ReusableObject ${this.id} created.`);
    }

    doWork() {
        this.log(`ReusableObject ${this.id} is working.`);
    }

    log(message) {
        const output = document.getElementById('output');
        const p = document.createElement('p');
        p.textContent = message;
        output.appendChild(p);
    }
}

File: Creational/Object Pool/C#/ReusableObject.cs
namespace ObjectPoolExample {
    public class ReusableObject {
        private static int _counter = 0;
        public int ID { get; private set; }

        public ReusableObject() {
            ID = ++_counter;
            Console.WriteLine($"ReusableObject {ID} created.");
        }

        public void DoWork() {
            Console.WriteLine($"ReusableObject {ID} is working.");
        }
    }
}
